File: 1338-reduce-array-size-to-the-half/1338-reduce-array-size-to-the-half.cpp
class Solution {
public:
    int minSetSize(vector<int>& arr) {
        int n=arr.size();
        unordered_map<int,int> mp;
        for(auto x:arr){
            mp[x]++;
        }
        vector<int> freq;
        for(auto [_,f]: mp){
            freq.push_back(f);
        }
        sort(freq.begin(),freq.end(),greater<int>());
        int ans=0;
        int half=n/2;
        int count=0;
        for(int i=0;i<n;i++){
            ans+=1;
            count+=freq[i];
            if (count>=half){
                break;
            }
                
        }
        return ans;
        
    }
};

File: 1338-reduce-array-size-to-the-half/1338-reduce-array-size-to-the-half.py
class Solution:
    def minSetSize(self, arr: List[int]) -> int:
        mp=dict()
        for i in arr:
            if i not in mp:
                mp[i]=1
            else:
                mp[i]+=1
        mp=dict(sorted(mp.items(), key=lambda item: -item[1]))
        count=0
        ans=0
        for i,j in mp.items():
            count+=j
            ans+=1
            if count>=(len(arr)//2):
                break
                
        return ans
        